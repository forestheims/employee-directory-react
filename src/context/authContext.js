import { createContext, useContext, useState } from 'react';
import { getUser } from '../services/auth';

const AuthContext = createContext();

export function AuthProvider({ children }) {
  const currentUser = getUser();

  const [user, setUser] = useState(
    currentUser
      ? { email: currentUser.email, id: currentUser.id }
      : { email: '', id: '' }
  );

  const contextValue = {
    user,
    setUser,
  };

  return (
    <AuthContext.Provider value={contextValue}>{children}</AuthContext.Provider>
  );
}

export function useAuth() {
  const context = useContext(AuthContext);
  if (context === undefined) {
    throw new Error('Error: useAuth needs to be called within a AuthProvider');
  }
  return context;
}
