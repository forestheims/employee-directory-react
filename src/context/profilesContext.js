import { createContext, useContext, useState, useEffect } from 'react';
import { getProfile } from '../services/profiles';
import { useAuth } from './authContext';

const ProfileContext = createContext();

export function ProfileProvider({ children }) {
  const [profile, setProfile] = useState({
    name: '',
    email: '',
    bio: '',
    birthday: '',
  });
  const { user } = useAuth();

  useEffect(() => {
    const fetchProfile = async () => {
      if (user.email) {
        try {
          const response = await getProfile();
          console.log(response);
          setProfile({
            name: response[0].name,
            email: user.email,
            bio: response[0].bio,
            birthday: response[0].birthday,
          });
        } catch (error) {
          console.log(error);
        }
      }
    };
    fetchProfile();
  }, [user.email]);

  const contextValue = { profile, setProfile };

  return (
    <ProfileContext.Provider value={contextValue}>
      {children}
    </ProfileContext.Provider>
  );
}

export function useProfile() {
  const context = useContext(ProfileContext);
  if (context === undefined) {
    throw new Error(
      'Error: useProfile needs to be called within a ProfileProvider'
    );
  }
  return context;
}
