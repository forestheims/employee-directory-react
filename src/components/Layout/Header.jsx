import React from 'react';
import { Link } from 'react-router-dom';
import ACMELogo from '../../assets/ACME_Logo.jpg';
import { useAuth } from '../../context/authContext';
import { useProfile } from '../../context/profilesContext';
import { signOutUser } from '../../services/auth';

export default function Header() {
  const { user, setUser } = useAuth();
  const { setProfile } = useProfile();

  const handleLogout = async () => {
    await signOutUser();
    setUser({ email: '', id: '' });
    setProfile({ email: '', name: '', bio: '', birthday: '' });
  };

  return (
    <header>
      <a href="https://villains.fandom.com/wiki/ACME">
        <img src={ACMELogo} alt="GitHub Logo" style={{ width: 100 }} />
      </a>
      <Link to="/">Home</Link>
      <Link to="/profile">Profile</Link>
      <Link to="/profile/add">Add Profile</Link>
      <Link to="/profile/edit">Edit Profile</Link>
      {user.email ? (
        <button onClick={handleLogout}>Logout</button>
      ) : (
        <Link to="/login">Sign In</Link>
      )}
      <h3>
        {user.email
          ? `${user.email} is Signed In`
          : 'Welcome. Sign in to access the directory.'}
      </h3>
    </header>
  );
}
