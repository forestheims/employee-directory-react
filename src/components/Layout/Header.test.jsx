import React from 'react';
import { render, screen } from '@testing-library/react';
import Header from './Header';
import { AuthProvider } from '../../context/authContext';
import { ProfileProvider } from '../../context/profilesContext';
import { MemoryRouter } from 'react-router-dom';

test('renders learn react link', async () => {
  const container = render(
    <AuthProvider>
      <ProfileProvider>
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      </ProfileProvider>
    </AuthProvider>
  );

  expect(container).toMatchSnapshot();
});
