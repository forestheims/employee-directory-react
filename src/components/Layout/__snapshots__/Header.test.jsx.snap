// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <a
          href="https://villains.fandom.com/wiki/ACME"
        >
          <img
            alt="GitHub Logo"
            src="/path/image"
            style="width: 100px;"
          />
        </a>
        <a
          href="/"
        >
          Home
        </a>
        <a
          href="/profile"
        >
          Profile
        </a>
        <a
          href="/profile/add"
        >
          Add Profile
        </a>
        <a
          href="/profile/edit"
        >
          Edit Profile
        </a>
        <a
          href="/login"
        >
          Sign In
        </a>
        <h3>
          Welcome. Sign in to access the directory.
        </h3>
      </header>
    </div>
  </body>,
  "container": <div>
    <header>
      <a
        href="https://villains.fandom.com/wiki/ACME"
      >
        <img
          alt="GitHub Logo"
          src="/path/image"
          style="width: 100px;"
        />
      </a>
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/profile"
      >
        Profile
      </a>
      <a
        href="/profile/add"
      >
        Add Profile
      </a>
      <a
        href="/profile/edit"
      >
        Edit Profile
      </a>
      <a
        href="/login"
      >
        Sign In
      </a>
      <h3>
        Welcome. Sign in to access the directory.
      </h3>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
